### PR TITLE
(PC-25124) docs(tests): add test strategy + examples

### PR DESCRIPTION
**Pull request by voisinhugo**
<https://passculture.atlassian.net/browse/PC-25124>

MSW config is inspired from: <https://github.com/vercel/next.js/tree/canary/examples/with-msw>

### CODE REVIEW

File: .vscode/perf-test.code-snippets [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      "\t\tawait screen.findByTestId('${2:testIDToWaitFor}')",
+      "\t  },",
+      "\t})",
+      "  })",
+      "})",
+      ""
+    ],
+    "description": "Create a performance test for a page"
+  }
+}
```
Comment (Lucasbeneston) [inline .vscode/perf-test.code-snippets:26]: Pas besoin de ce snippet

File: frontend/.vscode/a11y-test.code-snippets [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "React-Native Component": {
+    "prefix": "a11y-test",
+    "scope": "javascriptreact, typescriptreact",
+    "body": [
+      "import React from 'react'",
+      "",
+      "import { render, checkAccessibilityFor } from 'tests/utils'",
```
Comment (Lucasbeneston) [inline frontend/.vscode/a11y-test.code-snippets:8]:
```suggestion
      "import { render, checkAccessibilityFor } from 'tests/utils'",
```
Il n'est pas bon ce snippet

File: frontend/doc/development/unit-tests/component-test.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+it('should redirect to home page when signin is successful', async () => {
+  render(<Login />)
+  mockSignIn.mockReturnValueOnce(true)
+
+  const connexionButton = await screen.findByText('Se connecter')
+  fireEvent.press(connexionButton)
+
+  await waitFor(() => {
+    expect(navigate).toHaveBeenCalledTimes(1)
```
Comment (Lucasbeneston) [inline frontend/doc/development/unit-tests/component-test.md:33]: Nous n'avons pas de navigate ? Je pense qu'il faut supprimer toutes la doc qui ne nous sert pas encore, et l'ajouter petit a petit

Comment (Lucasbeneston) [reply frontend/doc/development/unit-tests/component-test.md:33]: Genre component-test ?

File: frontend/doc/development/unit-tests/general-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+**Storybook** is used when testing
+
+- a component that is used multiple times in the app
+- a component from the design system
+- a component which size differs depending on the device
+
+**Storybook** is not used when testing
+
+- a big and complex component
+- a component used in a single page
```
Comment (Lucasbeneston) [inline frontend/doc/development/unit-tests/general-guidelines.md:33]: Nous n'avons pas de storybook pour le moment

File: frontend/doc/development/unit-tests/general-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+## FIRST principles
+
+- Fast: Could be run anytime during the development phase
+- Independant/Isolated: Test should also follow 3A (Arrange, Act, Assert, ou Given-When-Then) and test one functionality
+- Repeatable: Each test sould give the same result each time it runs
+- Self-validating: The output of a test should be "yes" or "no", and not some values to check
+- Thorough: Tests must cover all possibilities and edge cases
+- Timely: Test should be written before their feature (TDD)
+
+## When to use storybook and snapshots
```
Comment (Lucasbeneston) [inline frontend/doc/development/unit-tests/general-guidelines.md:12]: Changer le titre du coup ?

File: frontend/src/pages/__snapshots__/about.test.tsx.snap [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+exports[`About page > should render the page 1`] = `
+<div>
+  <p
+    class="sc-iBdoyZ zlGK"
+  >
+    Chargement...
+  </p>
+</div>
+`;
```
Comment (Lucasbeneston) [inline frontend/src/pages/__snapshots__/about.test.tsx.snap:11]: Mauvais snapshot, il teste que ça charge, il faudrait ajouter msw et tester quand c'est une 200

Comment (voisinhugo) [reply frontend/src/pages/__snapshots__/about.test.tsx.snap:11]: Indeed, je fais ça 👍

File: package.json [hunk 7]
```diff
@@ -14,5 +14,10 @@
   },
   "dependencies": {
     "concurrently": "^8.2.2"
+  },
+  "devDependencies": {
+    "@testing-library/react": "^14.0.0",
+    "@types/jest-axe": "^3.5.7",
+    "jest-axe": "^8.0.0"
```
Comment (Lucasbeneston) [inline package.json:21]: Pas besoin de ça, encore moins des jest-quelqueChose

Comment (Lucasbeneston) [reply package.json:21]: C'est déjà dans `./frontend/package.json`

File: frontend/vitest.setup.ts [hunk 8]
```diff
@@ -4,6 +4,9 @@ import * as matchers from 'vitest-axe/matchers'
 
 import 'vitest'
 import 'vitest-canvas-mock'
+import { server } from '@/tests/server'
+
+server.listen()
```
Comment (marineshaw) [inline frontend/vitest.setup.ts:9]: Il faudrait pas avoir un server.close à la fin des tests aussi ? (Je sais que nous on a un close et un resetHandlers d'ailleurs)

**Review by marineshaw — APPROVED** [review]
